include/seqan3/range/view/translate_join.hpp

```diff
@@ -0,0 +1,405 @@
+                 std::Constructible<urng_t, ranges::ref_view<std::remove_reference_t<rng_t>>>
+    //!\endcond
+    view_translate_join(rng_t && _urange, translation_frames const _tf = translation_frames::SIX_FRAME)
+     : view_translate_join{std::view::all(std::forward<rng_t>(_urange)), _tf}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\overload
+    const_iterator begin() const noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\overload
+    const_iterator cbegin() const noexcept
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the container.
+     * \returns Iterator to the first element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() noexcept
+    {
+        return {*this, size()};
+    }
+
+    //!\overload
+    const_iterator end() const noexcept
+    {
+        return {*this, size()};
+    }
+
+    //!\overload
+    const_iterator cend() const noexcept
+    {
+        return end();
+    }
+    //!\}
+
+    /*!\brief Returns the number of elements in the view.
+     * \returns The number of elements in the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    size_type size() noexcept
+    {
+        return (size_type) seqan3::size(urange) * selected_frames.size();
+    }
+
+    //!\overload
+    size_type size() const noexcept
+    {
+        return (size_type) seqan3::size(urange) * selected_frames.size();
+    }
+
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the n-th element.
+     * \param[in] n The element to retrieve.
+     * \returns Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * Accessing an element behind the last causes undefined behaviour. In debug mode an assertion checks the size of
+     * the container.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (never modifies data).
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    reference operator[](size_type const n)
+    {
+        assert(n < size());
+        unsigned index_frame = n % selected_frames.size();
+        unsigned index_urange = (n - index_frame) / selected_frames.size();
```

Comment:
```suggestion
        size_type index_urange = (n - index_frame) / selected_frames.size();
```
Also in the line above.